src/adapter/src/coord.rs

```diff
@@ -2937,6 +2951,28 @@ impl Coordinator {
                         Message::CheckSchedulingPolicies
                     },
 
+                    // WIP(aljoscha): We could also do this inside the compute
+                    // controller, as part of "maintenance". But that also feels
+                    // slightly iffy.
```

Comment:
(can you say more about what feels iffy about having it here vs having it there?)

Reply:
Having the adapter layer manage the comms with the deployment widget seems good and sensible to me. And you've already pushed down the actual are-clusters-hydrated logic into the controller. So this seems good to me!

Reply:
I had slight reservations about adding yet another interval/branch to the coord main loop, but, as you can see, I chose to implement it this way. If you also feel good about it those reservations have been assuaged!